kuma/users/tests/test_templates.py

```diff
@@ -409,3 +411,263 @@ def test_common_reasons_empty(self):
+
+        self.client.login(username='admin', password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup_summary',
+                          kwargs={'user_id': testuser.id})
+        full_ban_url = self.client.get(ban_url)['Location']
+
+        resp = self.client.post(full_ban_url)
+        eq_(200, resp.status_code)
+        page = pq(resp.content)
+
+        revisions_reverted = page.find('#revisions-reverted li')
+        revisions_reverted_text = ''
+        for rev in revisions_reverted:
+            revisions_reverted_text += rev.text_content()
+
+        eq_(len(revisions_reverted), len(revisions_expected))
+        # The title for each of the created revisions shows up in the template
+        for revision in revisions_expected:
+            ok_(revision.title in revisions_reverted_text)
+        # The title for the original revision is not in the template
+        ok_(original_revision.title not in revisions_reverted_text)
+
+    def test_no_user_revisions_summary_page_template(self):
+        """If user has no revisions, it should be stated in summary template."""
+        testuser = self.user_model.objects.get(username='testuser')
+        admin = self.user_model.objects.get(username='admin')
+
+        # Create an original revision on a document by the admin user
+        document = create_document(save=True)
+        create_revision(
+            title='Revision 0',
+            document=document,
+            creator=admin,
+            save=True)
+
+        # The expected text
+        exp_reverted = "The user did not have any revisions that were reverted."
+        exp_followup = "The user did not have any revisions needing follow-up."
+
+        self.client.login(username='admin', password='testpass')
+        ban_url = reverse('users.ban_user_and_cleanup_summary',
+                          kwargs={'user_id': testuser.id})
+        full_ban_url = self.client.get(ban_url)['Location']
+
+        resp = self.client.post(full_ban_url)
+        eq_(200, resp.status_code)
+        page = pq(resp.content)
+
+        revisions_reverted = page.find('#revisions-reverted li')
+        revisions_followup = page.find('#revisions-followup li')
+        revisions_reverted_section = page.find('#revisions-reverted')
+        revisions_followup_section = page.find('#revisions-followup')
+
+        eq_(len(revisions_reverted), 0)
+        eq_(len(revisions_followup), 0)
+        ok_(exp_reverted in revisions_reverted_section.text())
+        ok_(exp_followup in revisions_followup_section.text())
+
+
+#    TODO: Phase III:
+#    def test_unreverted_changes_in_summary_page_template(self):
+
+
+class ProfileDetailTestCase(UserTestCase):
+    def test_user_profile_detail_ban_link(self):
+        """The user profile page for a user who has been banned"""
+        """should display correct text on ban links if user is banned or not banned"""
+        testuser = self.user_model.objects.get(username='testuser')
```

Comment:
Please change to this multi-line docstring format:

    """
    Test link text for banned and unbanned users.

    The user profile page for a user who has been banned should display
    correct  text on ban links if user is banned or not banned
    """

The format is:

    """
    The first line, ending with a period before column 80.

    After a blank space, a longer description if needed to understand
    the test case.
    """

In some verbose modes, docstrings are used instead of test names, and very long docstrings like this one will make the test output even uglier.  If you use this format, only the first line will be displayed in the verbose mode.

A long docstring sometimes suggests a test should be broken up. For example, this test could be broken up into a test for the link text for a banned user, and the link text for an unbanned user.

Also, thanks for using docstrings for the tests :smile:.  Another paragraph in my "how to style tests" docs in 2017...